### DISCUSSION TITLE
What is the better way to set a timeout on the transcribe method?

### DISCUSSION THREAD

**Discussion by ElvisFDias** [SYSTRAN/faster-whisper#264] [Q&A]
Hi!
I've created a web api to transcribe short audio files, and I need to set a timeout on the process transcribe. 
Someone know what's the better way to do this?
Thanks

**Comment by guillaumekln** [comment]
Hi,

The transcribe method returns a generator so a basic approach would be to exit the loop reading the segments. For example:

```python
segments, _ = model.transcribe("audio.mp3")

timeout = 10  # in seconds
start = time.time()

for segment in segments:
    print(segment)
    if time.time() - start >= timeout:
        break
```

However, with this method the transcription will not stop exactly at the configured timeout. It can only stop after generating a segment.
